src/cmakeProject.ts

```diff
@@ -498,15 +529,18 @@ export class CMakeProject {
     });
 
     private readonly sourceDirSub = this.workspaceContext.config.onChange('sourceDirectory', async () => {
```

Comment:
This doesn't look right.  I don't think the project should be listening for changes to `sourceDirectory`.  These changes should be listened for at the folder level.  The folder can then delete CMakeProjects that are no longer needed and add new ones when necessary.

Reply:
I added a listener at the folder level and missed removing this one. I think this was causing some errors.